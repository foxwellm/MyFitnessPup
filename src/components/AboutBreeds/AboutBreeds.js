import React, { Component } from 'react'
import { connect } from 'react-redux'
import BreedCard from '../BreedCard/BreedCard'
import { Link } from 'react-router-dom'
import trail from '../../assets/images/trail.jpg'
let shortID = require('short-id');

export class AboutBreeds extends Component {

  render() {
    const { staticBreeds, staticBreedInfo, location } = this.props
    const breedCards = staticBreeds.map((breed, i) => {
      return <Link className='BreedCard-link' to={`/about-breeds/${breed.breed}`} key={shortID.generate()}>
        <BreedCard
          {...breed}
          info={staticBreedInfo[breed.name]}
          location={location}
          active={false}
          cardNumber={i}
        />
      </Link>
    })

    return (
      <div className='breed-cards-container'>
        <img className='trail-img' alt='outdoor trail' src={trail} />
      {breedCards}
      </div >
    )
  }
}

export const mapStateToProps = (state) => ({
  staticBreeds: state.staticBreeds,
  staticBreedInfo: state.staticBreedInfo,
})

export default connect(mapStateToProps)(AboutBreeds)