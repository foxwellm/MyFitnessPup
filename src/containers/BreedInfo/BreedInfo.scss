.breed-info-container {
  background-color: rgba(0, 0, 0, 0.7);
  left: 0;
  top: 0;
  height: 100%;
  width: 100%;
  position: fixed;
} 

.form {
  background-color: $primary-light;
  font-family: $font-secondary;
  border: 1px solid black;
  padding: 20px;
  border-radius: 5px;
  // margin-left: 50%;
  // margin-top: 50%;
  left: 20vw;
  top: 20vh; 
  box-shadow: 2px 2px 6px black;
  width: 60vw;
  height: 60vh;
  position: fixed;
  // z-index: 1;
  // display: flex;
  // flex-direction: column;
  // max-height: 450px;
}