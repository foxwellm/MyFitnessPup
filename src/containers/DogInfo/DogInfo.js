import React, {Component} from 'react'

export class DogInfo extends Component {
  render() {
    return (
      <div>
        Dog Info
      </div>
    )
  }
}

export default DogInfo