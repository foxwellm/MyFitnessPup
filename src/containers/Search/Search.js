import React, { Component } from 'react'
import { fetchDogs } from '../../thunks/fetchDogs'
import { connect } from 'react-redux'
import { Switch, Route, withRouter } from 'react-router-dom'
import { breeds } from '../../staticData/breeds'

export class Search extends Component {
  constructor(props) {
    super(props)
    this.state = {
      zipCode: 77043,
      zipError: '',
      breeds: breeds,
      currentPage: 1,
      search: []
    }
  }


  handleChange = (e) => {

    this.setState({
      zipCode: e.target.value
    })
  }

  handleSearch = (e) => {
    e.preventDefault()
    const { search, breeds, zipCode } = this.state
    const { fetchDogs, storedDogs } = this.props

    if (search.length === 0) {

      breeds.forEach(dog => {
        if (!storedDogs[zipCode] || !storedDogs[zipCode][dog.breed]) {
          let options = {
            location: zipCode,
            // location: 77043,
            breed: dog.breed
          }
          fetchDogs(options)
        } else {
          console.log('there')
        }
      })
    }
  }

  render() {

    // const breeds = breeds
    // debugger

    return (
      <form onSubmit={this.handleSearch}>
        <input type='number' onChange={this.handleChange} placeholder='zip-code' name='zip' value={this.state.zipCode}></input>
        <button>Search</button>
        <div className='results-container'>
        
        </div>
      </form>
    )
  }
}

export const mapStateToProps = (state) => ({
  storedDogs: state.storedDogs
})

export const mapDispatchToProps = (dispatch) => ({
  fetchDogs: (options) => dispatch(fetchDogs(options))
})

export default connect(mapStateToProps, mapDispatchToProps)(Search)