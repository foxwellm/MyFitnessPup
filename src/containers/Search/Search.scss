.Search {
  height: calc(100% - 80px);
  display: flex;
  flex-direction: column;
}

.search-container {
  display: flex;
  overflow: scroll;
  height: 0px;
  transition: .15s;
  background-color: white;
}

.show {
  height: 200px;
  transition: .15s;
}

.trail-img {
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
}

.dog-loading {
  position: absolute;
  z-index: -1;
  top: 20%;
  left: 20%;
  height: 60%;
  width: 60%;
}

.search-bar {
  display: flex;
  justify-content: space-around;
  height: 30px;
  padding: 20px 0;
  position: relative;
  z-index: 2;
  background-color: white;
}

.results-container {
  flex: 1;
  position: relative;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(360px, 1fr));
  grid-auto-rows: auto;
  // grid-gap: 1%;
  width: 90%;
  padding: 0 150px 0px 0;
  margin: auto;
  overflow-y: scroll;
}

.search-buttons-container {
  width: 100%;
  display: flex;
  justify-content: flex-end;
}

.next-prev-btn {
    text-decoration: none;
  cursor: pointer;
  transition: background-color 0.1s ease-out;
  text-align: center;
  border: 0;
  margin:20px;
  font-size: 15px;
  font-family: "Lato", sans-serif;
  color: #fcfcfc;
  background-color: salmon;
  height: 35px;
  width: 80px;
  border-radius: 3px;
  outline: none;
  &:hover {
    transform: scale(1.01);
    background-color: rgb(232, 99, 84);
    transition: background-color 0.1s ease-out;
  }
}

.search-input-container {
  display: flex;
  position: relative;
}

.search-num-display {
  font-size: 20px;
}

.search-error {
  position: absolute;
  top: 5px;
  left: -250px;
  color: red;
  font-size: 20px;
}

.search-input {
  font-size: 20px;
  height: 30px;
  padding-left: 8px;
  width: 92px;
}

.search-btn {
  text-decoration: none;
  cursor: pointer;
  transition: background-color 0.1s ease-out;
  text-align: center;
  border: 0;
  margin: 0 0 15px 10px;
  font-size: 15px;
  font-family: "Lato", sans-serif;
  color: #fcfcfc;
  background-color: #107568;
  height: 35px;
  width: 80px;
  border-radius: 3px;
  outline: none;
  &:hover {
    transform: scale(1.01);
    background-color: #005b4f;
    transition: background-color 0.1s ease-out;
  }
}

.specific-search-btn {
  width: 200px;
}


