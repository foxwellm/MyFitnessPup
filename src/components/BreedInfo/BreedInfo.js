import React from 'react'
import AboutBreeds from '../AboutBreeds/AboutBreeds';
import { Link } from 'react-router-dom'

export const BreedInfo = (props) => {

  return (
    <div>
      <AboutBreeds />
      <div className='breed-info-container'>
        <div className='breed-info-card'>
          <div className='back-btn-container'>
            <Link to='/about-breeds'>
              <button className='breed-info-btn'>Back</button>
            </Link>
          </div>
          <p>{props.info.desc}</p>
          <div className='breed-info-img'>
            <img className='breed-img' alt='dog' src={require(`../../assets/images/${props.breed.img}`)} />
          </div>
        </div>
      </div>
    </div>
  )
}

export default BreedInfo