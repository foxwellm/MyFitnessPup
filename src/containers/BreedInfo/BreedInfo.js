import React, {Component} from 'react'
import AboutBreeds from '../../components/AboutBreeds/AboutBreeds';
import {Link} from 'react-router-dom'
import {importImages} from '../../helpers/importImages'

export class BreedInfo extends Component {

  render() {
    const {img} = this.props.breed
    const images = importImages();
    return (
      <div>
        <AboutBreeds />
        <div className='breed-info-container'>
        <div className='breed-info-card'>
            <div className='back-btn-container'><Link to='/about-breeds'><button className='breed-info-btn'>Back</button></Link></div>
            <p>{this.props.info.desc}</p>
            <div className='breed-info-img' style={{ backgroundImage: `url(${images[img]})` }}></div>
        </div>
        </div>
      </div>
    )
  }
}

export default BreedInfo