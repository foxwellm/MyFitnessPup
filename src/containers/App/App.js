import React, { Component } from 'react';
import { connect } from 'react-redux'
import Header from '../../components/Header/Header'
import Home from '../../components/Home/Home'
import AboutBreeds from '../../components/AboutBreeds/AboutBreeds'
import BreedInfo from '../../components/BreedInfo/BreedInfo'
import Search from '../Search/Search'
import DogInfo from '../../components/DogInfo/DogInfo';
import NotFound from '../../components/NotFound/NotFound'
import { Switch, Route, withRouter } from 'react-router-dom'
import { breeds, info } from '../../staticData/breeds'
import { storeStaticBreeds, storeStaticBreedInfo } from '../../actions'

export class App extends Component {

  componentDidMount = () => {
    this.props.storeStaticBreeds(breeds)
    this.props.storeStaticBreedInfo(info)
  }

  render() {
    const { staticBreeds, staticBreedInfo, fetchedDogs } = this.props
    return (
      <div className="App">
        <Header />
        <Switch>
          <Route exact path='/' component={Home} />
          <Route exact path='/about' render={() => <Home isAbout={true} />} />
          <Route exact path='/about-breeds' component={AboutBreeds} />
          <Route path='/about-breeds/:breed' render={({ match }) => {
            const { breed } = match.params
            const breedNames = staticBreeds.map(type => type.breed)
            const dogInfo = staticBreedInfo.find(dog => dog.name === breed)
            const dogBreed = staticBreeds.find(dog => dog.breed === breed)
            if (breedNames.includes(breed)) {
              return <BreedInfo breed={dogBreed} info={dogInfo} />
            }
            return <NotFound />
          }}
          />
          <Route path='/dog/:id' render={({ match }) => {
            let { id } = match.params
            id = parseInt(id)
            const dogFromStore = fetchedDogs.find(dog => dog.id === id)
            if (dogFromStore) {
              return <DogInfo dogFromStore={dogFromStore} />
            }
            return <NotFound />
          }}
          />
          <Route path='/search' component={Search} />
          <Route component={NotFound} />
        </Switch>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  staticBreeds: state.staticBreeds,
  staticBreedInfo: state.staticBreedInfo,
  fetchedDogs: state.fetchedDogs,
})

export const mapDispatchToProps = (dispatch) => ({
  storeStaticBreeds: (breeds) => dispatch(storeStaticBreeds(breeds)),
  storeStaticBreedInfo: (info) => dispatch(storeStaticBreedInfo(info)),
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App))
