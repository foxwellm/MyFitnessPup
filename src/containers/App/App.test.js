import React from 'react';
import ReactDOM from 'react-dom';
import { App } from './App';
import { Home } from '../Home/Home'
import { Search } from '../Search/Search'
import { AboutBreeds } from '../../components/AboutBreeds/AboutBreeds'
import { BreedInfo } from '../BreedInfo/BreedInfo'
import { NotFound } from '../../components/NotFound/NotFound'
import {DogInfo} from '../DogInfo/DogInfo'
import { shallow, mount } from 'enzyme'
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension'
import { rootReducer } from '../../reducers/index';
import thunk from 'redux-thunk';
import { breeds } from '../../staticData/breeds'
// import BreedCard from '../../components/BreedCard/BreedCard';

describe('App', () => {
  const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)))
  let wrapper

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('should match the correct snapshot', () => {
    wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('Routes', () => {

    it('should render the Home container when at the root route', () => {
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(Home)).toHaveLength(1)
    })

    it.skip('should render the AboutBreeds container when at /about-breeds', () => {
      // const require = jest.fn()
      // require.context() = jest.fn().mockImplementation(() => 5)
      const mockBreeds = breeds

      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/about-breeds']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(AboutBreeds)).toHaveLength(1)
    })

    it('should render the BreedInfo container when at /about-breeds/:breed and given a valid breed', () => {
      const mockBreeds = breeds
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/about-breeds/Poodle']}>
            <App breeds={mockBreeds} />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(BreedInfo)).toHaveLength(1)
    })

    it('should render the NotFound container when at /about-breeds/:breed and given an invalid breed', () => {
      const mockBreeds = breeds
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/about-breeds/Chihuahua']}>
            <App breeds={mockBreeds} />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(NotFound)).toHaveLength(1)
    })

    it('should render the Search container when at /search', () => {
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/search']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(Search)).toHaveLength(1)
    })

    it.skip('should render the DogInfo container when at /search/:id and given a valid id', () => {
      const mockBreeds = breeds
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/search/N3456']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(DogInfo)).toHaveLength(1)
    })

    it.skip('should render the NotFound container when at /search/:id and given an invalid id', () => {
      const mockBreeds = breeds
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/search/ZZ1']}>
            <App breeds={mockBreeds} />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(NotFound)).toHaveLength(1)
    })
  })

})