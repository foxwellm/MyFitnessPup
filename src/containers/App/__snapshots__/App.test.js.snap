// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the correct snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about-breeds"
    />
    <Route
      path="/about-breeds/:breed"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/search"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
